examples/se_2_3_localization.cpp

```diff
@@ -0,0 +1,452 @@
+                             " Vel: " << X.linearVelocity().transpose() << endl;
+    cout << "X unfilterd : Pos: " << X_unfiltered.translation().transpose() <<
+                             " RPY: " << X_unfiltered.rotation().eulerAngles(2, 1, 0).reverse().transpose() <<
+                             " Vel: " << X_unfiltered.linearVelocity().transpose() << endl;
+    cout << "---------------------------------------------------------------------------" << endl;
+    // END DEBUG
+
+
+    // START TEMPORAL LOOP
+    //
+    //
+
+    // Make 10 steps. Measure up to three landmarks each time.
+    for (double t = 0; t < 10; t+=dt)
+    {
+        //// I. Simulation ###############################################################################
+
+        /// get current simulated state and measurements from previous step
+        auto R_k = X_simulation.rotation();
+        auto v_k = X_simulation.linearVelocity();
+        auto acc_k = alpha_prev + R_k.transpose()*g;
+
+        /// input vector
+        u_nom << (dt*(R_k.transpose())*v_k + 0.5*dt*dt*acc_k),  dt*omega_prev, dt*acc_k;
+
+        /// simulate noise
+        u_noise = u_sigmas * Array9d::Random();             // control noise
+        u_noisy = u_nom + u_noise;                          // noisy control
+
+        u_simu   = u_nom;
+        u_unfilt = u_noisy;
+
+        /// first we move - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+        X_simulation = X_simulation + u_simu;               // overloaded X.rplus(u) = X * exp(u)
+        /// update expected IMU measurements
+        alpha = alpha_const - X_simulation.rotation().transpose()*g; // update expected IMU measurement after moving
+
+        /// then we measure all landmarks - - - - - - - - - - - - - - - - - - - -
+        for (int i = 0; i < landmarks.size(); i++)
+        {
+            b = landmarks[i];                               // lmk coordinates in world frame
+
+            /// simulate noise
+            y_noise = y_sigmas * Array3d::Random();         // measurement noise
+
+            y = X_simulation.inverse().act(b);              // landmark measurement, before adding noise
+            y = y + y_noise;                                // landmark measurement, noisy
+            measurements[i] << y;                     // store for the estimator just below
+        }
+
+
+
+
+        //// II. Estimation ###############################################################################
+
+        /// get current state estimate to build the state-dependent control vector
+        auto R_k_est = X.rotation();
+        auto v_k_est = X.linearVelocity();
+        auto acc_k_est = alpha_prev + R_k_est.transpose()*g;
+
+        Eigen::Vector3d accLin = dt*(R_k_est.transpose())*v_k_est + 0.5*dt*dt*acc_k_est;
+        Eigen::Vector3d gLin = R_k_est.transpose()*g*dt;
+        Eigen::Matrix3d accLinCross = manif::skew(accLin);
+        Eigen::Matrix3d gCross = manif::skew(gLin);
```

Comment:
Same for these lines. Can we use the Lie group operators directly on the object X and not breaking it into pieces?

Reply:
Since this is the Jacobian of a state-dependent input motion vector with respect to a state perturbation vector, we do not already have some methods that would give us this Jacobian. 

Which is why I had split it into these equations.
It also depends on the input vector we might define. (The above is specific for a strap down IMU based motion model).

Reply:
OK. Check my comment below.